lib/spack/spack/package_base.py

```diff
@@ -1886,7 +1886,10 @@ def _setup_test(self, verbose, externals):
             pkg_id = self.test_suite.test_pkg_id(self.spec)
         else:
             self.test_log_file = fsys.join_path(self.stage.path, _spack_install_test_log)
+            self.test_suite = TestSuite([self.spec])
+            self.test_suite.stage = self.stage.path
```

Comment:
:+1: to this in the short term, at least we treat `test_suite` as other test related attributes